src/EditorFeatures/Core/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -1,22 +1,73 @@
-﻿using Microsoft.CodeAnalysis.FindUsages;
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
 using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
 
 namespace Microsoft.CodeAnalysis.Editor.SymbolSearch
 {
-    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile
+    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile, IResultHasClassifiedDefinition, IResultHasClassifiedContext //, IResultOpensWithCustomAction
     {
         private DefinitionItem definition;
+        private SourceReferenceItem reference;
+        private RoslynSymbolSource source;
+        private static ISymbolOriginDefinition LocalSymbolOrigin = null;
 
-        public RoslynSymbolResult(DefinitionItem definition)
+        public RoslynSymbolResult(RoslynSymbolSource source, DefinitionItem definition)
         {
+            this.source = source;
             this.definition = definition;
+            Owner = source;
+            Name = definition.ToString();
+            Origin = GetLocalOrigin();
+            ClassifiedDefinition = new ClassifiedTextElement(definition.NameDisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            ClassifiedContext = new ClassifiedTextElement(definition.DisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            var sampleLocation = definition.SourceSpans.First(); // TODO: handle all locations
+            PersistentSpan = source.ServiceProvider.PersistentSpanFactory.Create(sampleLocation.Document.FilePath, AsSpan(sampleLocation.SourceSpan), SpanTrackingMode.EdgeInclusive);
         }
 
-        public IPersistentSpan PersistentSpan => throw new System.NotImplementedException();
+        public RoslynSymbolResult(RoslynSymbolSource source, SourceReferenceItem reference)
+            : this(source, reference.Definition)
+        {
+            this.reference = reference;
+            Name = reference.ToString(); // which one runs first?
+        }
+
+        private Span AsSpan(TextSpan sourceSpan)
+        {
+            return new Span(sourceSpan.Start, sourceSpan.Length);
+        }
+
+        private ISymbolOriginDefinition GetLocalOrigin()
+        {
+            if (LocalSymbolOrigin == null)
+            {
+                source.ServiceProvider.SymbolSearchBroker.SymbolOriginDefinitions.TryGetValue(PredefinedSymbolOrigins.LocalCode, out LocalSymbolOrigin);
+            }
+            return LocalSymbolOrigin;
```

Comment:
1. what's going on here?
2. what are the threading guarantees for this method?
3. can this be static?